FinSearchUnified/Tests/Helper/UrlBuilderTest.php

```diff
@@ -83,4 +101,99 @@ private function setClientIp($ipAddress)
     {
         $_SERVER['HTTP_CLIENT_IP'] = $ipAddress;
     }
+
+    /**
+     * @return array
+     */
+    public function forceOriginalQueryProvider()
+    {
+        return [
+            'forceOriginalQuery not present' => [null],
+            'forceOriginalQuery present and truthy' => [1],
+            'forceOriginalQuery present and falsy' => [0],
+        ];
+    }
+
+    /**
+     * @dataProvider forceOriginalQueryProvider
+     *
+     * @param int|null $forceOriginalQuery
+     *
+     * @throws \Exception
+     */
+    public function testBuildUrlAndResponse($forceOriginalQuery)
+    {
+        $sessionArray = [
+            ['isSearchPage', true]
+        ];
+        // Create mock object for Shopware Session and explicitly return the values
+        $session = $this->getMockBuilder(Enlight_Components_Session_Namespace::class)
+            ->setMethods(['offsetGet'])
+            ->getMock();
+        $session->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($sessionArray);
+
+        // Assign mocked session variable to application container
+        Shopware()->Container()->set('session', $session);
+
+        $_GET[Constants::SDYM_PARAM_FORCE_QUERY] = $forceOriginalQuery;
```

Comment:
Use the string instead of the constant here. Imagine that we change its value but accidentally add a typo while doing so. We would never notice that we are sending the wrong parameter since we are using the constant everywhere.

So, constants are good. Use them in the code where applicable but still consider using strings in tests instead of constants. This way we will notice soon enough if the parameter name was changed to something wrong.

Also change other occurrences of `Constants::SDYM_PARAM_FORCE_QUERY` in this test.